mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/fragment/OtpFragment.kt

```diff
@@ -42,31 +44,23 @@ class OtpFragment : BaseFragment() {
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
-        val rootView = inflater.inflate(R.layout.fragment_otp, container, false) as ViewGroup
-        ButterKnife.bind(this, rootView)
-        mPeOtp!!.setOnEditorActionListener(OnEditorActionListener { v, actionId, event ->
-            if (actionId == EditorInfo.IME_ACTION_DONE) {
-                okayClicked()
-                return@OnEditorActionListener true
-            }
-            false
-        })
-        mPeOtp!!.requestFocus()
-        return rootView
-    }
 
-    fun okayClicked() {
-        if (activity is SetupUpiPinActivity) {
-            if (mPeOtp!!.text.toString() == otp) {
-                (activity as SetupUpiPinActivity?)!!.otpVerified()
-            } else {
-                showToast(getString(R.string.wrong_otp))
+        return ComposeView(requireContext()).apply {
+            setContent {
+                MifosTheme {
+                    OtpScreen(realOtp =otp ?: "",
+                        onOtpTextCorrectlyEntered ={
+                            onOtpTextCorrectlyEntered()
+                        })
+                }
             }
         }
     }
 
-    fun showToast(message: String?) {
-        Toaster.showToast(activity, message)
+    fun onOtpTextCorrectlyEntered() {
+        if (activity is SetupUpiPinActivity) {
+            (activity as SetupUpiPinActivity?)!!.otpVerified()
```

Comment:
use require activity

Reply:
changed the logic . so it is no longer required